sandbox: fills are generated by internally aggregated bars instead of quote/ticks

# Bug Report

### Expected Behavior
Context: sandbox paper trading with Bybit.
The strategy is subscribed to quote ticks and trade ticks. Internally aggregated bars should not generate fills.

### Actual Behavior
A fill is generated based on an internally aggregated bar.

```
2024-05-16T13:13:47.194220001Z [INFO] PAPER-TRADING-001.ReinforcementLearningMarketMakerStrategy: <--[EVT] OrderAccepted(instrument_id=ETHUSDT-LINEAR.BYBIT, client_order_id=O-20240516-1313-001-000-37484, venue_order_id=BYBIT-679-29793, account_id=BYBIT-001, ts_event=1715865227193603001)
2024-05-16T13:13:47.203959001Z [DEBUG] PAPER-TRADING-001.OrderMatchingEngine(BYBIT): Processing QuoteTick(ETHUSDT-LINEAR.BYBIT,2970.62,2970.63,19.10,9.98,1715865227132999936)
2024-05-16T13:13:47.213644001Z [DEBUG] PAPER-TRADING-001.OrderMatchingEngine(BYBIT): Processing QuoteTick(ETHUSDT-LINEAR.BYBIT,2970.62,2970.63,20.78,12.70,1715865227143000064)
2024-05-16T13:13:47.224158001Z [DEBUG] PAPER-TRADING-001.OrderMatchingEngine(BYBIT): Processing QuoteTick(ETHUSDT-LINEAR.BYBIT,2970.62,2970.63,20.78,14.50,1715865227153999872)
2024-05-16T13:13:47.225103001Z [DEBUG] PAPER-TRADING-001.OrderMatchingEngine(BYBIT): Processing Bar(ETHUSDT-LINEAR.BYBIT-1-SECOND-MID-INTERNAL,2971.220,2971.220,2970.625,2970.625,928.31,1715865227000000000)
2024-05-16T13:13:47.225182001Z [DEBUG] PAPER-TRADING-001.OrderMatchingEngine(BYBIT): Applying fills to LimitOrder(SELL 0.03 ETHUSDT-LINEAR.BYBIT LIMIT @ 2_971.22 GTC, status=ACCEPTED, client_order_id=O-20240516-1313-001-000-37484, venue_order_id=BYBIT-679-29793, position_id=None, tags=None), venue_position_id=None, position=None, fills=[(Price(2971.220), Quantity(0.03))]
2024-05-16T13:13:47.227014001Z [ERROR] PAPER-TRADING-001.DataEngine: RuntimeError: Invalid price precision for fill 3 when instrument price precision is 2. Check that the data price precision matches the ETHUSDT-LINEAR.BYBIT instrument
2024-05-16T13:13:47.227219001Z [DEBUG] PAPER-TRADING-001.DataEngine: Data message queue stopped
```

### Steps to Reproduce the Problem

1. sandbox paper trading with Bybit
2. Subscribe to quote ticks, trade ticks and 1-SECOND-MID-INTERNAL internal bars.
3.

### Specifications

- OS platform: MacOS
- Python version: 3.11.4
- `nautilus_trader` version: develop branch, git commit 3046583810824c44c8271e34a7c30aa06cffcec2

--- a/nautilus_trader/adapters/sandbox/config.py
+++ b/nautilus_trader/adapters/sandbox/config.py
@@ -28,9 +28,12 @@ class SandboxExecutionClientConfig(LiveExecClientConfig, frozen=True, kw_only=Tr
         The currency for this venue
     balance : int
         The starting balance for this venue
+    bar_execution: bool
+        If bars should be processed by the matching engine(s) (and move the market).
 
     """
 
     venue: str
     currency: str
     balance: int
+    bar_execution: bool = True

--- a/nautilus_trader/adapters/sandbox/execution.py
+++ b/nautilus_trader/adapters/sandbox/execution.py
@@ -87,6 +87,7 @@ class SandboxExecutionClient(LiveExecutionClient):
         oms_type: OmsType = OmsType.NETTING,
         account_type: AccountType = AccountType.MARGIN,
         default_leverage: Decimal = Decimal(10),
+        bar_execution: bool = True,
     ) -> None:
         self._currency = Currency.from_str(currency)
         money = Money(value=balance, currency=self._currency)
@@ -124,6 +125,7 @@ class SandboxExecutionClient(LiveExecutionClient):
             fee_model=MakerTakerFeeModel(),
             latency_model=LatencyModel(0),
             clock=self.test_clock,
+            bar_execution=bar_execution,
             frozen_account=True,  # <-- Freezing account
         )
         self._client = BacktestExecClient(

--- a/nautilus_trader/adapters/sandbox/factory.py
+++ b/nautilus_trader/adapters/sandbox/factory.py
@@ -73,5 +73,6 @@ class SandboxLiveExecClientFactory(LiveExecClientFactory):
             venue=name or config.venue,
             balance=config.balance,
             currency=config.currency,
+            bar_execution=config.bar_execution,
         )
         return exec_client

